src/llmq/chainlocks.cpp

```diff
@@ -519,7 +519,11 @@ MessageProcessingResult CChainLocksHandler::HandleNewRecoveredSig(const llmq::CR
             // already got the same or a better CLSIG through the CLSIG message
             return {};
         }
-
+        const auto pindex = m_chainstate.m_chain.Tip()->GetAncestor(lastSignedHeight);
```

Comment:
if we will have a new chainlock we **need** to switch chain here to the new one, because it is a new chain-locked tip and current tip should be refused, isn't it true?